python/data_structures/linked_list.py

```diff
@@ -37,6 +42,65 @@ def includes(self, value):
 
         return False
 
+    def append(self, new_value):
+        """
+        set your your new node as an attribute to the new value of the Node class.
+        self.head is the current node we are on
+        the value of the head will be set to the value of the new node
+        while that value is none, the value will become the value of the next node which is now the value of the end node.
+        """
+        end_node = Node(new_value)
+        if self.head is None:
+            self.head = end_node
+            return
+        else:
+            held_value = self.head
+            while held_value.next is not None:
+                held_value = held_value.next
+            held_value.next = end_node
+
+    def insert_before(self, value, new_value):
+        """
+        make a new node
+        iterate through the list looking for the element previous to it
+        we can re-assign next to
+
+        """
+        new_node = Node(new_value)
+        # fine the node we want to insert before
+        # save value from previous node and check next value if next node is not before value, reassign temp
+        # repeat
+        # if next value is not before then don't reassign temp
+        current = self.head
+        if self.head is None:
+            raise TargetError
+        if self.includes(value) is False:
+            raise TargetError
+        if current.value is value:
+            new_node.next = self.head
+            self.head = new_node
```

Comment:
lines 79 - 81 match 83 - 85 word for word. In Coding, try to abide by the DRY principle, which stands for Don't Repeat Yourself.

Would the code still function properly if 79 - 81 were removed completely?